collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java

```diff
@@ -2098,17 +2092,6 @@ public void loadingComplete(FormLoaderTask task, FormDef formDef, String warning
                 }
             }
 
-            boolean pendingActivityResult = task.hasPendingActivityResult();
-
-            if (pendingActivityResult) {
```

Comment:
I believe this mechanism was used to handle receiving activity results like for example images (ImageWidget) after process death?
Now (even without this change but using the current master) if you have no background processes set in `Developer options` the result (image in this case) is ignored and the form is reloaded which means we end up in the hierarchy view. As I remember in the past it was possible to receive activity results in such a case. The changes you have implemented in that area recently changed that and I don't know if that was a deliberate decision.
I'm afraid that when using some older devices with little memory it might be difficult to receive some activity results when switching to other apps is required like when taking photos because then the form is reloaded and we are moved to the hierarchy view and the result is ignored.

Reply:
> Now (even without this change but using the current master) if you have no background processes set in Developer options the result (image in this case) is ignored and the form is reloaded which means we end up in the hierarchy view. As I remember in the past it was possible to receive activity results in such a case. The changes you have implemented in that area recently changed that and I don't know if that was a deliberate decision.

I tried this and it still seems to work. The steps I ran through were:

1. Set "Background process limit" to "No background processes"
2. Open a form with an image widget
3. Click "Take picture"
4. Take a picture in the camera app

In this case in both v2023.1.2 and v2023.2.2 I return to Collect with the picture. Are you seeing different results? I had thought you were probably correct though and that the Collect process would be in "the background" while the Camera is doing it's thing. Maybe Android keeps processes waiting for results in the foreground? Maybe this differs between API levels if that's the case?

Reply:
I think in my scenario there was one extra step which was switching to another app after opening the camera app. So it would be:

1. Set "Background process limit" to "No background processes"
2. Open a form with an image widget
3. Click "Take picture"
4. Switch to another app and get back to the camera app
5. Take a picture in the camera app

Reply:
Ah I see. That flow is definitely not working any longer (it is in v2022.4.4). I can switch this PR from removing that code, to getting it working again!